content/docs/connect/choose-connection.md

```diff
@@ -0,0 +1,163 @@
+
+### Step 2: If serverless, choose HTTP or WebSocket connections
+
+Neon offers two types of serverless connections: HTTP and WebSocket.
+
+- **Use HTTP for**: One-shot queries and short-lived operations where connection setup and teardown costs are minimal.
+- **Use WebSocket for**: [Transactions](/docs/postgresql/query-reference#transactions) and operations requiring persistent connections over a session. WebSocket connections provide lower latency for multiple queries in the same session.
+
+<Admonition type="note">
+We are working on automatic switching between HTTP and WebSocket to as needed. Check our [roadmap](/docs/introduction/roadmap) to see what's coming soon and our Friday [Changelog](/docs/changelog) for the features-of-the-week.
+</Admonition>
+
+### Step 3: Next, choose your connection type: direct or pooled
+
+You then need to decide whether to use direct connections or pooled connections using PgBouncer:
+
+- **In general, use pooled connections whenever you can**: Pooled connections can efficiently manage high numbers of concurrent connections, supporting up to 10,000 connection. This 10,000 connection limit is most useful for serverless applications and application-side connection pools that have many open connections, but infrequent and/or short transactions.
+- **Use direct (unpooled) connections if you need persistent connections**: If your application is focused mainly on tasks like migrations or administrative operations that require stable and long-lived connections, use an unpooled connection.
+
+<Admonition type="note">
+The use of connection pooling is not a magic bullet. PgBouncer has a `default_pool_size` of 64, meaning it can handle only 64 direct connections to the database at any one time. When your application connects to PgBouncer, it joins a large pool, but only 64 connections are processed at a time. Long operations can cause these connections to time out or close before tasks are completed.
+</Admonition>
+
+For more information on these choices, see:
+
+- [Neon Serverless Driver](/docs/serverless/serverless-driver)
+- [Connection pooling](/docs/connect/connection-pooling)
+
+## Common Pitfalls
+
+Here are some key points to help you navigate potential issues.
+| Issue | Description |
+|----------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Double Pooling | Avoid using client-side pooling if you're using PgBouncer on the Neon side. Just let Neon handle the pooling to prevent retaining unused connections on the client side. If you must use client-side pooling, make sure it releases connections back to the pool early enough to avoid conflicts with PgBouncer. |
```

Comment:
```suggestion
| Double Pooling | Avoid using client-side pooling if you're using a pooled Neon connection (supported by PgBouncer). Just let Neon handle the pooling to prevent retaining unused connections on the client side. If you must use client-side pooling, make sure connections are released back to the client-side pool early enough to avoid conflicts with PgBouncer. |
```